lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb

```diff
@@ -337,6 +337,7 @@ def cmd_background_help
 
   def cmd_background
     print_status("Backgrounding session #{client.name}...")
+    Readline::HISTORY.length.times {Readline::HISTORY.pop}
```

Comment:
It seems here like this should be popping the context from the stack and deferring the history update to the history manager's `#pop_context` method.

Reply:
Changed :+1: